### PR TITLE
[qtquick3d] devendor meshoptimizer

### PR DESCRIPTION
**Pull request by m-kuhn**
@Neumann-A does this need to be upstreamed to be acceptable?

### CODE REVIEW

**Comment by FrankXie05** [comment]
This patch is too large. 🤨

**Comment by m-kuhn** [comment]
I could reduce the patch and not remove the library (or use `file(REMOVE_RECURSE src/3rdparty/meshoptimizer)` instead)

**Comment by FrankXie05** [comment]
We can download it through `vcpkg_download_distfile`. 
eg: https://github.com/microsoft/vcpkg/blob/c8582b4d83dbd36e1bebc08bf166b5eb807996b0/ports/alsa/portfile.cmake#L14

**Comment by m-kuhn** [comment]
where from?

**Comment by m-kuhn** [comment]
let me reduce it, I think it's more transparent and maintainable

File: ports/qtquick3d/0001-devendor-meshoptimizer.patch [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+-        ../3rdparty/meshoptimizer/src/vcacheanalyzer.cpp
+-        ../3rdparty/meshoptimizer/src/vcacheoptimizer.cpp
+-        ../3rdparty/meshoptimizer/src/vertexcodec.cpp
+-        ../3rdparty/meshoptimizer/src/vertexfilter.cpp
+-        ../3rdparty/meshoptimizer/src/vfetchanalyzer.cpp
+-        ../3rdparty/meshoptimizer/src/vfetchoptimizer.cpp
+     DEFINES
+         QT_BUILD_QUICK3DUTILS_LIB
+     INCLUDE_DIRECTORIES
+         ../3rdparty/xatlas
```
Comment (Neumann-A) [inline ports/qtquick3d/0001-devendor-meshoptimizer.patch:55]: xatlas should probably also be devendored then?

Comment (m-kuhn) [reply ports/qtquick3d/0001-devendor-meshoptimizer.patch:55]: probably, I only did the one that was causing pain for me and xatlas would require a new port afaics.

**Comment by Neumann-A** [comment]
> @Neumann-A does this need to be upstreamed to be acceptable?

Why are you asking me. Having it upstream would be better for future updates. You probably need to move the find_package logic into a configure.cmake file which has a feature depending on `INPUT_<something>='system'`. qtmultimedia has a few of those if i remember correctly.

**Comment by m-kuhn** [comment]
> > @Neumann-A does this need to be upstreamed to be acceptable?
> 
> Why are you asking me.

For this kind of valuable feedback:

> Having it upstream would be better for future updates. You probably need to move the find_package logic into a configure.cmake file which has a feature depending on `INPUT_<something>='system'`. qtmultimedia has a few of those if i remember correctly.

**Comment by m-kuhn** [comment]
Upstream PR here https://codereview.qt-project.org/c/qt/qtquick3d/+/599071/1

**Comment by m-kuhn** [comment]
@FrankXie05 does devendoring require upstream approval?

**Review by FrankXie05 — DISMISSED** [review]

**Comment by FrankXie05** [comment]
> @FrankXie05 does devendoring require upstream approval?

It looks like the upstream has not responded for a long time. We can do this here.

**Pull request merged**